engine/src/main/java/org/terasology/rendering/opengl/fbms/DynamicFBOsManager.java

```diff
@@ -0,0 +1,133 @@
+import org.terasology.rendering.oculusVr.OculusVrHelper;
+import org.terasology.rendering.opengl.AbstractFBOsManager;
+import org.terasology.rendering.opengl.DefaultDynamicFBOs;
+import static org.terasology.rendering.opengl.DefaultDynamicFBOs.FINAL;
+import static org.terasology.rendering.opengl.DefaultDynamicFBOs.READ_ONLY_GBUFFER;
+import static org.terasology.rendering.opengl.DefaultDynamicFBOs.WRITE_ONLY_GBUFFER;
+import org.terasology.rendering.opengl.FBO;
+import org.terasology.rendering.opengl.FBOConfig;
+import org.terasology.rendering.opengl.ScreenGrabber;
+
+/**
+ * TODO: Add javadocs
+ * TODO: Better naming
+ */
+public class DynamicFBOsManager extends AbstractFBOsManager {
+    private FBO.Dimensions fullScale;
+    private RenderingConfig renderingConfig;
+    private ScreenGrabber screenGrabber;
+
+    public DynamicFBOsManager(Context context) {
+        renderingConfig = context.get(Config.class).getRendering();
+        screenGrabber = context.get(ScreenGrabber.class);
+        fullScale = new FBO.Dimensions(Display.getWidth(), Display.getHeight());
+        generateDefaultFBOs();
+    }
+
+    @Override
+    public void request(FBOConfig fboConfig) {
+        ResourceUrn fboName = fboConfig.getName();
+        if (fboConfigs.containsKey(fboName)) {
+            if (!fboConfig.equals(fboConfigs.get(fboName))) {
+                throw new IllegalArgumentException("Requested FBO is already available with different configuration");
+            }
+        } else {
+            generateWithDimensions(fboConfig, fullScale.multiplyBy(fboConfig.getScale()));
+        }
+        retain(fboName);
+    }
+
+    private void generateDefaultFBOs() {
+        generateDefaultFBO(READ_ONLY_GBUFFER);
+        generateDefaultFBO(WRITE_ONLY_GBUFFER);
+        generateDefaultFBO(FINAL);
+    }
+
+    private void generateDefaultFBO(DefaultDynamicFBOs defaultDynamicFBO) {
+        FBOConfig fboConfig = defaultDynamicFBO.getConfig();
+        FBO fbo = generateWithDimensions(fboConfig, fullScale.multiplyBy(fboConfig.getScale()));
+        defaultDynamicFBO.setFbo(fbo);
+    }
+
+    /**
+     * Invoked before real-rendering starts
+     * TODO: how about completely removing this, and make Display observable and this FBM as an observer
+     */
+    public void update() {
+        updateFullScale();
+        if (get(READ_ONLY_GBUFFER.getName()).dimensions().areDifferentFrom(fullScale)) {
+            disposeAllFBOs();
+            createFBOs();
+            updateDefaultFBOs();
+        }
+    }
+
+    private void updateDefaultFBOs() {
+        READ_ONLY_GBUFFER.setFbo(fboLookup.get(READ_ONLY_GBUFFER.getName()));
+        WRITE_ONLY_GBUFFER.setFbo(fboLookup.get(WRITE_ONLY_GBUFFER.getName()));
+        FINAL.setFbo(fboLookup.get(FINAL.getName()));
+    }
+
+    private void disposeAllFBOs() {
+        for (ResourceUrn urn : fboConfigs.keySet()) {
+            fboLookup.get(urn).dispose();
+            fboLookup.remove(urn);
```

Comment:
No need to do this if you do fboLookup.clear() at the end of the loop I think.
In fact, I wonder if removing a map entry while iterating the map is even possible without generating an error.